src/VisualStudio/Core/Def/Implementation/Workspace/PersistentStorageService.cs

```diff
@@ -119,6 +146,38 @@ public IPersistentStorage GetStorage(Solution solution)
             }
         }
 
+        private bool ShouldUseEsent(Solution solution)
+        {
+            if (_testing)
+            {
+                return true;
+            }
+
+            // we only use esent for primary solution. (Ex, forked solution will not use esent)
+            if (solution.BranchId != solution.Workspace.PrimaryBranchId || solution.FilePath == null)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool SolutionSizeAboveThreshold(Solution solution)
+        {
+            if (_testing)
+            {
+                return true;
+            }
+
+            if (_solutionSizeTracker == null)
+            {
+                return false;
+            }
+
+            var size = _solutionSizeTracker.GetSolutionSize(solution.Workspace, solution.Id);
+            return size > SolutionSizeThreshold;
```

Comment:
```return _testing || _solutionSizeTracker?.GetSolutionSize(solution.Workspace, solution.Id) > SolutionSizeThreshold;```

(There is an active issue on the compiler to ensure that the last part of this creates optimal code.)